MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm

```diff
@@ -618,14 +610,19 @@ static CALayerContentsGravity getCALayerContentsGravity(VkSwapchainPresentScalin
 		_presentableImages.push_back(_device->createPresentableSwapchainImage(&imgInfo, this, imgIdx, nullptr));
 	}
 
-	NSString* screenName = @"Main Screen";
+	auto* mtlLayer = getCAMetalLayer();
+	if (mtlLayer) {
+		NSString* screenName = @"Main Screen";
 #if MVK_MACOS && !MVK_MACCAT
-	if ([mtlLayer.screenMVK respondsToSelector:@selector(localizedName)]) {
-		screenName = mtlLayer.screenMVK.localizedName;
-	}
+		if ([mtlLayer.screenMVK respondsToSelector:@selector(localizedName)]) {
```

Comment:
And this call *definitely* needs to happen on the main thread, because that's the one that triggered the crash the reporter saw.

Reply:
Ditto.